solved_exercises/dani/04-functions.py

```diff
@@ -79,4 +76,143 @@ def is_prime(number):
     return all_mod_multiplied > 0
 
 
-print("Primes list up to", value, "is", primes(value))
+def test_factorial():
+    """we'll test the correctnes of factorial function"""
+
+    print("Testing factorial() function...")
+
+    test_data = [
+        # (value, expected)
+        (-1, 1),
```

Comment:
This is not mathematically correct. The factorial of negative numbers is a complex number. Let's stick to positive integers only.